### PR TITLE
News Section

### PR DESCRIPTION
**Pull request by ElectronicsArchiver**
<br>

*I hope this is enough of a description,*
*if not, just ask me ..*

<br>
<br>

As mentioned in the **[Package Discussion](https://github.com/orgs/pulsar-edit/discussions/107#discussioncomment-4615076)**:

I thought it would be nice to have a small news section
on GitHub itself, considering many people don't use the
social media channels you use to post about small updates.

*I for example don't use Discord.*

*And as mentioned, on the blog page of the website isn't much*

These sections could also be more developer focused since it's on GitHub!
news

<br>

## Preview

The following link takes you to the rendered preview,
**however**, the actual organization overview is not
as wide, so it fits better there.

*( Could be adjusted anyways )*

https://github.com/ElectronicsArchiver/.github/tree/main/profile

<br>
<br>

## Features / Structure

`.github/News/Data/News.yaml` Contains news article data
↳ `.github/.github/workflows/News.yaml` Workflow starting the updater
↳ `.github/profile/README.md` Inserts new section

The workflow can be manually triggered but also watches the `/News/Data/News.yaml` file.

The updater is:

- [A small Deno program](https://github.com/ElectronicsArchiver/.github/tree/main/News/Source)

- [Has properly scoped permissions](https://github.com/ElectronicsArchiver/.github/blob/main/News/deno.json) 

- [Has set standard library file versions](https://github.com/ElectronicsArchiver/.github/blob/main/News/Imports.json)

- Written in TypeScript

<br>

The [`News.yaml`](https://github.com/ElectronicsArchiver/.github/blob/main/News/Data/News.yaml) file has a comment header with an explanation
on how to use it, as well as a commented out example, ..

News articles can contain:

- A title ( bold )
- Multiple lines of content
- Links 
  
  Uses the GitHub markdown like syntax of `[This] is a link`
  where the link is placed in a link list, as seen in the
  example in the comment.

<br>
<br>

## Injection

The news section is inserted into the README 
by searching for the following marker:

```markdown
<!---------------------------------- 🗞 News ---------------------------------->
```

When updating, everthing between 2 of these
markers is replaced with the updated content.

<br>
<br>

## Maintenance

- As everything is versioned, the updater can't break over time

- There is no input validation, however since only staff can
  directly edit it anyways, this would just be a generic
  configuration issue.

  *I could still add it if you do care to have it,*
  *in the worst case, the README just doesn't update.*

- You can ping me if you want more features

- If there is interest, I might create a standalone
  project ( at some point ), then it would just be 
  an integration.

<br>

### CODE REVIEW

**Comment by ElectronicsArchiver** [comment]
@confused-Techie Finished it.

**Review by confused-Techie — CHANGES_REQUESTED** [review]
Honestly this does look pretty rad. We have the space to spare on the Readme of the org, so this doesn't seem like a bad solution, at least until we find other methods.

But I do want to just include a few notes:
* I know this is a repeat conversation, and that you feel strongly about it, but for the `.editorconfig` while it would be cool to start including it, I really think it would be a requirement that Pulsar natively supports that sort of thing first. Because otherwise it just highlights that some developers or that the Pulsar team as a whole isn't using Pulsar to make Pulsar. And as for me who doesn't have anything that can read `.editorconfig` it'd be just useless file bloat. And since this is also the position everyone took on another PR, hate to say would you mind removing that file?
* Also the Typescript is an interesting choice, don't think we have much of it around in the repo, especially since we are trying to standardize with more JavaScript. But that is to also say I don't think we could quite run a NodeJS app within an action like this without a dedicated builder, or with including our `node_modules` folder, so in this case it's a totally logical decision and is fine by me, especially since it's smaller and hopefully easily maintainable.

Otherwise for this one I'd love to hear what others from @pulsar-edit/documentation think about adding this feature into our ORG Readme?

If you'd like to see the conversation that caused this PR just read through [here](https://github.com/orgs/pulsar-edit/discussions/107)

**Comment by ElectronicsArchiver** [comment]
@confused-Techie On the editorconfig topic, sure I can remove it, 
but the point about it not native being supported by Pulsar doesn't 
seem sensible considering no editor does.

The point is, basically any somewhat remarkable editor has a plugin 
that does support it, just check their downloads page:

https://editorconfig.org/#download

**Comment by Daeraxa** [comment]
My main concern is that we are either splitting up information too much or making it hard for ourselves to keep things up to date.

We obviously have Discord announcements, we should be using Mastodon to mirror those same announcements and I guess this "news" section should also be one? Personally I'd prefer for it to be on the main site, maybe where we currently put the "notices" but if we include this here *and* we want the website then really we need some kind of system to publish to all these places from one location.

**Comment by confused-Techie** [comment]
@Daeraxa I totally agree we want to make our life easier to keep things up to date.

Something I'm thinking and I think mentioned in the linked discussion, is we could look into a Discord bot, to take our announcements channel and build an RSS feed from it. If we were to do that, then ideally we could have some way to make Reddit, Mastodon, and ideally this, or the website (like you mentioned) or both, all read from that RSS feed, so that one the announcements section is the one source of everything

**Comment by ElectronicsArchiver** [comment]
@Daeraxa Well if you make a system, for the news section it would simply have to update the `News.yaml` and the rest would be done automatically, so theres that.

**Comment by kaosine** [comment]
> My main concern is that we are either splitting up information too much or making it hard for ourselves to keep things up to date.
> 
> We obviously have Discord announcements, we should be using Mastodon to mirror those same announcements and I guess this "news" section should also be one? Personally I'd prefer for it to be on the main site, maybe where we currently put the "notices" but if we include this here *and* we want the website then really we need some kind of system to publish to all these places from one location.

Yea all our news sources should just mirror each other tbh. This sounds like a job for a bot with some kind of dashboard that admins can access to post various things too. That way it sends stuff out to mastodon, this, discord, and reddit. (along with a appropriate tag for the author if it's not a automated release update somehow)

**Comment by ElectronicsArchiver** [comment]
Btw, I'm currently working on making this a GitHub Action, just so you know.

**Comment by ElectronicsArchiver** [comment]
<br>

@confused-Techie Finally got around
to finishing the **[GitHub Action]** for this.

I still have to write the documentation,
though the example **[input]**, **[config]** &
**[workflow]** do show how to use it.

*Next, I will switch it out on this PR.*

<br>

[GitHub Action]: https://github.com/marketplace/actions/news-section
[Input]: https://github.com/MarkedDown/News/blob/main/Example/News.yaml
[Config]: https://github.com/MarkedDown/News/blob/main/.github/.News.yml 
[Workflow]: https://github.com/MarkedDown/News/blob/main/.github/workflows/Action_Test.yml

**Comment by ElectronicsArchiver** [comment]
The workflow / action seems to work fine.

**Review by confused-Techie — APPROVED** [review]
Love the use of an action here to keep this repos code short and sweet.

While I know the previous discussion stopped at wanting to find a proper system to sync up our information sources, lets be honest it's been over two weeks with zero progress on that front, other than I've built out a Webhook translator for Discord and GitHub for some of the unsupported events.

So Honestly I'm 100% fine with adding this and backlogging the previous messages we have sent.

I really do appreciate the effort on this one, and think it looks good!

> Also wanted to say appreciate being receptive to other suggestions and requests.

**Comment by ElectronicsArchiver** [comment]
So, .. it's approved, .. when will it be merged?

**Comment by confused-Techie** [comment]
> So, .. it's approved, .. when will it be merged?

Generally we like to wait for two approvals on a PR, where if that doesn't happen then I can go ahead and merge anyway. Essentially I'd like to wait a day or two to let others review and point out any issues they may have found that I missed or to provide their approval at which point for smaller changes like this we should be good to merge right away

File: profile/README.md
Comment (confused-Techie) [inline profile/README.md:43]:
```suggestion

```

Just gonna clear this all out to avoid commiting any incorrect messages

**Review by confused-Techie — APPROVED** [review]
Alright, well looks like since this one hasn't been seeing much activity I feel comfortable merging this for now.

I'll also go ahead and remove any of the output from the readme for the time being and make a PR against this repo with our updated information.

Thanks again for the contribution!

File: profile/README.md
Comment (confused-Techie) [reply profile/README.md:43]: Looks like I'm not able to commit just the removal. 

If you'd don't mind I'll make a commit against your PR branch to remove it, then merge the changes.

Comment (confused-Techie) [reply profile/README.md:43]: Or it won't quite let me do that either.

Guess I'll just merge it and right away make a PR to merge the proper text.

Comment (ElectronicsArchiver) [reply profile/README.md:43]: Yes, well by default I turn off commit access for my PRs,
since you already merged it, it doesn't matter anymore but
I can enable it if you need it next time.

Comment (Daeraxa) [reply profile/README.md:43]: I think it has been reverted, have you looked at #118?

**Pull request merged**
